File: python/high-scores/high_scores.py
class HighScores(object):
    def __init__(self, scores):
        self.scores = scores
    
    def latest(self):
        return self.scores[-1]

    def personal_best(self):
        return max(self.scores)

    def personal_top(self):
        return sorted(self.scores, reverse=True)[:3]
    
    def report(self):
        latest = self.scores[-1]
        best = max(self.scores)
        str1 = 'Your latest score was {latest}.'
    if latest == best:
            return str1 + ' That\'s your personal best!'
        return str1 + f' That\'s {best - latest} short of your personal best!'

File: javascript/high-scores/high-scores.js
export class HighScores {
  constructor(scores) {
    this.scores = scores;
    this.latest = scores[scores.length - 1];
    this.highest = Math.max(...scores);
    this.getTopScores();
    this.getReport();
  }

  getTopScores() {
    this.top = [];
    const temp = this.scores.slice();
    for (let i = 0; i < this.scores.length && i < 3; i++) {
      this.top.push(temp.splice(temp.findIndex(t => t === Math.max(...temp)), 1)[0]);
    }
  }

  getReport() {
    const diff = this.highest - this.latest;
    this.report = `Your latest score was ${this.latest}. That's${diff > 0 ? (
      ` ${diff} short of`
     ) : ''} your personal best!`;
  }
}